### PR TITLE
slight model re-factoring

### PR DESCRIPTION
**Pull request by andenixa**
- removed excess threading code
  - added random kernel initialization to dense layer

added (optional) DDSIM loss function (requires keras_contrib package) which allegedly should give a better perceptual training quality

### CODE REVIEW

**Comment by andenixa** [comment]
Is it going to be merged?
